src/ngx_http_lua_regex.c

```diff
@@ -1922,12 +1924,78 @@ ngx_http_lua_ngx_re_sub_helper(lua_State *L, unsigned global)
 }
 
 
+ngx_int_t
+ngx_http_lua_set_jit_stack_size(int size)
+{
+    ngx_http_lua_main_conf_t    *lmcf;
+
+    lmcf = ngx_http_cycle_get_module_main_conf(ngx_cycle,
+                                               ngx_http_lua_module);
+
+    if (lmcf->regex_cache_entries > 0) {
+        return -1;
+    }
+
+    if (size < NGX_LUA_RE_MIN_JIT_STACK_SIZE) {
+        size = NGX_LUA_RE_MIN_JIT_STACK_SIZE;
+    }
+
+    if (lmcf->jit_stack) {
+        pcre_jit_stack_free(lmcf->jit_stack);
```

Comment:
We use it in `init_by_lua` and have not seen any noticeable memory leaks across HUP reloads. That's not to say that we aren't leaking but we have not noticed it. Anyways, how can we solve it? I genuinely do not know enough about nginx's memory pools and layout, esp. across reloads.